modules/graph/src/endpoints/vulnerability.rs

```diff
@@ -32,3 +83,125 @@ pub async fn affected_products(
 ) -> actix_web::Result<impl Responder> {
     Ok(HttpResponse::Ok().finish())
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::graph::Graph;
+    use actix_web::test::TestRequest;
+    use actix_web::web::Data;
+    use actix_web::{test, web, App};
+    use std::sync::Arc;
+    use trustify_common::db::Database;
+    use trustify_common::model::PaginatedResults;
+    use trustify_entity::vulnerability;
+
+    #[actix_web::test]
+    async fn all_vulnerabilities() -> Result<(), anyhow::Error> {
+        let db = Database::for_test("api_all_vulnerabilities").await?;
+        let graph = Arc::new(Graph::new(db));
+
+        let vuln = graph.ingest_vulnerability("CVE-111", ()).await?;
+
+        vuln.set_title(Some("Bob's CVE".to_string()), ()).await?;
+
+        let vuln = graph.ingest_vulnerability("CVE-222", ()).await?;
+
+        vuln.set_title(Some("Jim's CVE".to_string()), ()).await?;
+
+        let vuln = graph.ingest_vulnerability("CVE-333", ()).await?;
+
+        vuln.set_title(Some("Jenß CVE".to_string()), ()).await?;
+
+        let app = actix_web::test::init_service(
+            App::new()
+                .app_data(web::Data::from(graph.clone()))
+                .configure(crate::endpoints::configure),
+        )
+        .await;
+
+        let uri = "/api/v1/vulnerability";
+
+        let request = TestRequest::get().uri(uri).to_request();
+
+        let response: PaginatedResults<vulnerability::Model> =
+            test::call_and_read_body_json(&app, request).await;
+
+        assert_eq!(3, response.items.len());
+
+        Ok(())
+    }
+
+    #[actix_web::test]
+    async fn all_vulnerabilities_paginated() -> Result<(), anyhow::Error> {
+        let db = Database::for_test("api_all_vulnerabilities_paginated").await?;
+        let graph = Arc::new(Graph::new(db));
+
+        let vuln = graph.ingest_vulnerability("CVE-111", ()).await?;
+
+        vuln.set_title(Some("Bob's CVE".to_string()), ()).await?;
+
+        let vuln = graph.ingest_vulnerability("CVE-222", ()).await?;
+
+        vuln.set_title(Some("Jim's CVE".to_string()), ()).await?;
+
+        let vuln = graph.ingest_vulnerability("CVE-333", ()).await?;
+
+        vuln.set_title(Some("Jenß CVE".to_string()), ()).await?;
+
+        let app = actix_web::test::init_service(
+            App::new()
+                .app_data(web::Data::from(graph.clone()))
+                .configure(crate::endpoints::configure),
+        )
+        .await;
+
+        let uri = "/api/v1/vulnerability?limit=2";
+
+        let request = TestRequest::get().uri(uri).to_request();
+
+        let response: PaginatedResults<vulnerability::Model> =
+            test::call_and_read_body_json(&app, request).await;
+
+        assert_eq!(2, response.items.len());
+
+        Ok(())
+    }
+
+    #[actix_web::test]
```

Comment:
I'm really digging the `test-log` crate you found. Can you change this to `#[test_log::test(actix::test)]` so you can change the `println!` below to `log::debug!` maybe?